packages/ozone/src/daemon/blob-diverter.ts

```diff
@@ -0,0 +1,229 @@
+      serviceConfig: BlobReportServiceConfig
+    },
+  ) {
+    this.serviceConfig = services.serviceConfig
+    this.idResolver = services.idResolver
+  }
+
+  start() {
+    this.poll(this.pollState, () => this.divertBlob())
+  }
+
+  poll(state: PollState, fn: () => Promise<void>) {
+    if (this.destroyed) return
+    state.promise = fn()
+      .catch((err) => {
+        dbLogger.error({ err }, 'blob divert failed')
+      })
+      .finally(() => {
+        state.timer = setTimeout(() => this.poll(state, fn), 30 * SECOND)
+      })
+  }
+
+  async processAll() {
+    await Promise.all([this.divertBlob(), this.pollState.promise])
+  }
+
+  async destroy() {
+    this.destroyed = true
+    const destroyState = (state: PollState) => {
+      if (state.timer) {
+        clearTimeout(state.timer)
+      }
+      return state.promise
+    }
+    await destroyState(this.pollState)
+  }
+
+  async divertBlob() {
+    const toPush = await this.db.db
+      .selectFrom('blob_divert_event')
+      .select('id')
+      .forUpdate()
+      .skipLocked()
+      .where('divertedAt', 'is', null)
+      .where('attempts', '<', 10)
+      .execute()
+    await Promise.all(toPush.map((evt) => this.attemptBlobDivert(evt.id)))
+  }
+
+  private async getBlob({
+    pds,
+    did,
+    cid,
+  }: {
+    pds: string
+    did: string
+    cid: string
+  }) {
+    const blobResponse = await axios.get(
+      `${pds}/xrpc/com.atproto.sync.getBlob`,
+      {
+        params: { did, cid },
+        decompress: true,
+        responseType: 'stream',
+        timeout: 5000, // 5sec of inactivity on the connection
+      },
+    )
+    const imageStream: Readable = blobResponse.data
+    const verifyCid = new VerifyCidTransform(CID.parse(cid))
+    forwardStreamErrors(imageStream, verifyCid)
+
+    return {
+      contentType:
+        blobResponse.headers['content-type'] || 'application/octet-stream',
+      imageStream: imageStream.pipe(verifyCid),
+    }
+  }
+
+  private async uploadBlob(
+    {
+      imageStream,
+      contentType,
+    }: { imageStream: Readable; contentType: string },
+    { subjectDid, subjectUri }: { subjectDid: string; subjectUri: string },
+  ) {
+    if (!this.serviceConfig.authToken || !this.serviceConfig.url) {
+      return false
+    }
+
+    const url = `${this.serviceConfig.url}?did=${subjectDid}&uri=${subjectUri}`
+    const result = await axios(url, {
+      method: 'POST',
+      data: imageStream,
+      headers: {
+        Authorization: this.serviceConfig.authToken,
+        'Content-Type': contentType,
+      },
+    })
+
+    return result.status === 200
+  }
+
+  private async uploadBlobOnService({
+    subjectDid,
+    subjectUri,
+    subjectBlobCid,
+  }: {
+    subjectDid: string
+    subjectUri: string
+    subjectBlobCid: string
+  }): Promise<boolean> {
+    try {
+      if (!this.serviceConfig.authToken || !this.serviceConfig.url) {
+        throw new Error('Blob divert service not configured')
+      }
```

Comment:
If there's no service config, can we avoid creating a `BlobDiverter` at all?  It is a bit awkward to have to check both `serviceConfig.authToken` and `serviceConfig.url` in each method and opens the question about how each method should behave if they aren't present.